src/SFML/Audio/SoundFileWriterOpus.cpp

```diff
@@ -0,0 +1,254 @@
+            err() << "Possibly wrong sample rate, allowed are 8000, 12000, 16000, 24000, or 48000 Hz." << std::endl;
+        close();
+        return false;
+    }
+
+    m_file.open(filename.c_str(), std::ios::binary);
+    if (!m_file)
+    {
+        err() << "Failed to write ogg/opus file \"" << filename << "\" (cannot open file)" << std::endl;
+        close();
+        return false;
+    }
+
+    // Set bitrate (VBR is default)
+    opus_encoder_ctl(m_opus, OPUS_SET_BITRATE(128000));
+
+    // Create opus header MAGICBYTES VERSION
+    std::string magicBytes("OpusHead1");
+    std::vector<unsigned char> headerData(magicBytes.begin(), magicBytes.end());
+
+    headerData.push_back(channelCount);
+    writeUint32(headerData, static_cast<Uint32>(sampleRate));
+    headerData.push_back(channelCount > 8 ? 255 : (channelCount > 2)); // Mapping family
+
+    // Map opus header to ogg packet
+    ogg_packet op;
+    op.packet = &headerData.front(); // C++11: headerData.data();
+    op.bytes = headerData.size();
+    op.b_o_s = 1;
+    op.e_o_s = 0;
+    op.granulepos = 0;
+    op.packetno = m_packageNumber++;
+
+    // Write the header packet to the ogg stream
+    ogg_stream_packetin(&m_ogg, &op);
+    flushBlocks();
+
+    // Create comment header, needs to be in a new page
+    // commentData initialized with magic bytes
+    magicBytes = "OpusTags";
+    std::vector<unsigned char> commentData(magicBytes.begin(), magicBytes.end()); // C++11: commentData({'O', 'p', 'u', 's', 'T', 'a', 'g', 's'});
+
+    // Vendor string
+    const std::string opusVersion(opus_get_version_string());
+
+    // unsigned 32bit integer: Length of vendor string (encoding library)
+    writeUint32(commentData, opusVersion.size());
+    commentData.insert(commentData.end(), opusVersion.begin(), opusVersion.end());
+
+    // Length of user comments (E.g. you can add a ENCODER tag for SFML)
+    writeUint32(commentData, 0);
+
+    op.packet     = &commentData.front();
+    op.bytes      = commentData.size();
+    op.b_o_s      = 0;
+    op.e_o_s      = 0;
+    op.granulepos = 0;
+    op.packetno   = m_packageNumber++;
+    ogg_stream_packetin(&m_ogg, &op);
+
+    // This ensures the actual audio data will start on a new page, as per spec
+    flushBlocks();
+
+    return true;
+}
+
+
+////////////////////////////////////////////////////////////
+void SoundFileWriterOpus::write(const Int16* samples, Uint64 count)
+{
+    assert(m_opus);
+
+    const opus_uint32 frame_size = 960;
+    const opus_int32 buffer_size = frame_size * m_channelCount * sizeof(Int16);
+    unsigned char* buffer = new unsigned char[buffer_size];
+
+    Uint32 frame_number = 0;
+    Uint8 endOfStream = 0;
+
+    while (count > 0)
+    {
+        ogg_packet op;
+        opus_int32 packet_size;
+
+        // Check if wee need to pad the input
+        if (count < frame_size * m_channelCount)
+        {
+            const Uint32 begin = (frame_number * frame_size * m_channelCount);
```

Comment:
Unneeded parentheses (also in several other snippets). Generally, use operator precedence to your advantage, and add parentheses only where it actually clarifies code or where the code style demands it (for `||` and `&&` operators).